projects/plugins/jetpack/extensions/blocks/map/edit.js

```diff
@@ -223,14 +236,15 @@ const MapEdit = ( {
 
 	let content;
 
-	if ( preview ) {
+	if ( preview || isPreviewMode ) {
 		const mapStyleObject = styles.find( styleObject => styleObject.name === mapStyle );
 
 		content = (
 			<div>
 				<img
 					alt={ __( 'Map Preview', 'jetpack' ) }
 					src={ mapStyleObject ? mapStyleObject.preview : previewPlaceholder }
+					style={ { width: '100%', maxHeight: '400px', objectFit: 'cover' } }
```

Comment:
This extra style unexpectedly changes the height of the map in the preview.

This is the hover preview, which uses the `preview` attribute from [the block's example data](https://arc.net/l/quote/ngxlkfad).

| Before | After |
| --- | --- |
| ![CleanShot 2024-10-15 at 15 36 39](https://github.com/user-attachments/assets/4caae30a-46eb-41d5-b838-d3c7ef4d8fa8) | ![CleanShot 2024-10-15 at 12 49 51@2x](https://github.com/user-attachments/assets/3acb608d-4da6-4a86-862a-94f875617f40) |

It also changes the appearance of the pattern previews. The way they looked before matches the aspect ratio of the map when it is inserted into the page. So the preview was more accurate the way it was.

| Before | After |
| --- | --- |
| ![CleanShot 2024-10-15 at 12 52 35@2x](https://github.com/user-attachments/assets/493d7c06-3c92-477a-ab6e-afe16febd0b2) | ![CleanShot 2024-10-15 at 12 54 14@2x](https://github.com/user-attachments/assets/d5728257-6c84-42e1-9f99-d54380dc5f94) |

Maybe something like this will fix it?

```suggestion
					style={ { width: '100%', maxHeight: mapHeight ?? 400, objectFit: 'cover' } }
```

Reply:
Actually looks like it has to be this. Something about adding those styles changes the way the hover preview works for some reason.

```suggestion
					style={ preview ? null : { width: '100%', height: mapHeight ?? 400, objectFit: 'cover' } }
```